ouroboros-consensus-test/test-consensus/Test/Consensus/Mempool/Fairness.hs

```diff
@@ -0,0 +1,207 @@
+    let runConcurrently = Async.runConcurrently . asum . fmap Async.Concurrently
+    txs <- runConcurrently [                      adders  mempool smallTxSize
+                           , threadDelay 1_000 >> adders  mempool largeTxSize
+                           , threadDelay 1_000 >> remover mempool             nrOftxsToCollect
+                           ]
+    -- We want to give the small adders the head start, to make sure the mempool
+    -- is filled up with small transactions first. That is why we use
+    -- threadDelay above.
+
+    ----------------------------------------------------------------------------
+    --  Count the small and large transactions
+    ----------------------------------------------------------------------------
+    let
+      nrSmall :: Double
+      nrLarge :: Double
+      (nrSmall, nrLarge) = (fromIntegral . length *** fromIntegral . length)
+                         $ List.partition  (<= smallTxSize)
+                         $ fmap txSize txs
+    length txs @?= nrOftxsToCollect
+    -- At the end of the tests the proportion of small and large
+    -- transactions that were added should be rouhgly the same. We tolerate
+    -- the given theshold.
+    let diff_ratio = abs (nrLarge - nrSmall) / (nrLarge + nrSmall)
+    diff_ratio <= toleranceThreshold
+      @? (    "The difference between the number of large and small transactions added "
+           <> "exeeds the threshold (" <> show toleranceThreshold <> ")\n"
+           <> "Total number of small transactions that were added: " <> show nrSmall <> "\n"
+           <> "Total number of large transactions that were added: " <> show nrLarge <> "\n"
+           <> "Difference / Total: " <> show diff_ratio
+         )
+
+-- | The fairness test depends on the:
+--
+-- - Total mempool capacity.
+-- - Size of what we consider to be a small transaction.
+-- - Size of what we consider to be a large transaction.
+-- - How many added transactions we count.
+--
+-- The smaller the difference between 'smallTxSize' and 'largeTxSize', the
+-- harder it will be detect potential differences in way the mempool handles
+-- small and large transactions.
+--
+-- The larger the capacity, the higher the chance large transactions can be
+-- added before the mempool is saturated.
+--
+-- We tolerate a certain ratio between the difference of small and large
+-- transactions added, and the total transactions that were added. For instance,
+-- given a threshold of 0.2, if we measure the sizes of 100 added transactions,
+-- the difference between the number small and large transactions we counted
+-- should not be larger than 20.
+--
+data TestParams = TestParams {
+    mempoolMaxCapacity :: Word32
+  , smallTxSize        :: Word32
+  , largeTxSize        :: Word32
+  , nrOftxsToCollect   :: Int
+  , toleranceThreshold :: Double
+  }
+
+-- | Add transactions with with the specified size to the mempool.
+--
+-- We launch a fixed number of adder threads, set to 2 at the moment.
```

Comment:
`[0..2]` bounds are inclusives so it seems we are launching 3 threads, not 2. I would suggest to remove the precision about the number of threads in the comment.